### PR TITLE
plots: shorten names in the legend

### PR DESCRIPTION
**Pull request by dberenbaum**
Before:

visualization(24)

After:

visualization(25)

### CODE REVIEW

**Review by shcheklein — APPROVED** [review]

**Comment by dberenbaum** [comment]
Need to check VS Code and Studio

**Comment by dberenbaum** [comment]
Ping @ssachkovskaya. Could you check this doesn't break anything in Studio when you have a chance?

**Comment by dberenbaum** [comment]
@ssachkovskaya @yathomasi Would you like to check on how this looks in Studio before merge?

**Comment by yathomasi** [comment]
> this doesn't seem to break anything for VS Code but also doesn't help with its legend because the field for the legend is a separate field `filename::field` that is generated within VS Code

Screenshot 2023-02-24 at 15 03 09

For my checks, it seems to be the same case as VS Code for Studio as well. 

> If we can bring this logic into DVC, we can improve the plots across products.

Yes, now we use the same field `filename::field` in Studio as well.

**Comment by dberenbaum** [comment]
Great, thanks @yathomasi! Just wanted to make sure it doesn't make anything worse for now.

**Comment by dberenbaum** [comment]
@iterative/dvc This one should be ready to review and merge 🙏

File: dvc/render/converter/vega.py [hunk 1]
```diff
@@ -221,14 +222,24 @@ def flat_datapoints(self, revision):  # noqa: C901, PLR0912
...
+            all_y_fields = set()
 
         # override to unified y field name if there are different y fields
         if len(all_y_fields) > 1:
             props_update["y"] = "dvc_inferred_y_value"
         else:
             props_update["y"] = first(all_y_fields)
 
+        # get common prefix to drop from field and file names
+        common_field_prefix = _get_common_prefix(all_y_fields)
```
Comment (daavoo) [inline dvc/render/converter/vega.py:240]: To make sure I get it right, `field` would be the name actually stored inside the plot, right?

I see the point for `filename`, but why also alter `field`? 
Was the motivation coming from DVCLive storing the full paths in `field`?

Comment (dberenbaum) [reply dvc/render/converter/vega.py:240]: > Was the motivation coming from DVCLive storing the full paths in `field`?

Yes, at least partially, although a little hard to remember at this point 😅 . It still could be long if people use field names like `test/acc` on their own 🤔 .

Now that you bring it up, it's no longer an issue for DVCLive, and it's probably better to not mess with the user-provided field name. I think it makes sense, so I can update to drop altering the field.

Comment (daavoo) [reply dvc/render/converter/vega.py:240]: > Now that you bring it up, it's no longer an issue for DVCLive, and it's probably better to not mess with the user-provided field name. I think it makes sense, so I can update to drop altering the field.

Yep, that was where I was heading to 👍

Comment (dberenbaum) [reply dvc/render/converter/vega.py]: I figured 😄 . Should be updated with those changes now.

**Review by daavoo — APPROVED** [review]

**Pull request merged**